framework/PostProcessors/MCSimporter.py

```diff
@@ -194,3 +171,36 @@ def collectOutput(self, finishedJob, output):
       output.load(outputDict['data'], style='dict', dims=outputDict['dims'])
     else:
         self.raiseAnError(RuntimeError, 'MCSImporter failed: Output type ' + str(output.type) + ' is not supported.')
+
+def mcsReader(mcsListFile):
+  """
+    Function designed to read a file containing the set of MCSs and to save it as list of list
+    @ In, mcsListFile, string, name of the file containing the set of MCSs
+    @ Out, mcsIDs, np array, array containing the ID associated to each MCS
+    @ Out, probability, np array, array containing the probability associated to each MCS
+    @ Out, mcsList, list, list of MCS, each element is also a list containing the basic events of each MCS
+    @ Out, beList, list, list of all basic events contained in the MCSs
+  """
+  mcsList=[]
+  beList=set()
+  probability = np.zeros((0))
+  mcsIDs = np.zeros((0))
+
+  # construct the list of MCSs and the list of BE
+  with open(mcsListFile.getFilename(), 'r') as file:
+    next(file) # skip header
+    lines = file.read().splitlines()
+    for l in lines:
+      elementsList = l.split(',')
+
+      mcsIDs = np.append(mcsIDs,elementsList[0])
+      elementsList.pop(0)
+
+      probability=np.append(probability,elementsList[0])
+      elementsList.pop(0)
+
+      mcsList.append(list(element.rstrip('\n') for element in elementsList))
+
+      beList.update(elementsList)
+
+  return mcsIDs, probability, mcsList, beList
```

Comment:
did you try something like pandas? maybe it can avoid the loop (just a try)

https://stackoverflow.com/questions/39812493/dealing-with-missing-data-in-pandas-read-csv

Reply:
I have tired just a bit tbh, but  some files were formatted in a weird way that I have preferred a line by line parsing.